app/src/main/java/com/onegini/mobile/exampleapp/network/fcm/FCMInstanceIdService.java

```diff
@@ -16,35 +16,41 @@
 
 package com.onegini.mobile.exampleapp.network.fcm;
 
+import android.util.Log;
+import com.google.firebase.iid.FirebaseInstanceId;
 import com.google.firebase.iid.FirebaseInstanceIdService;
 import com.onegini.mobile.exampleapp.util.DeregistrationUtil;
-import com.onegini.mobile.sdk.android.handlers.OneginiMobileAuthWithPushEnrollmentHandler;
-import com.onegini.mobile.sdk.android.handlers.error.OneginiMobileAuthWithPushEnrollmentError;
+import com.onegini.mobile.sdk.android.handlers.OneginiRefreshMobileAuthPushTokenHandler;
+import com.onegini.mobile.sdk.android.handlers.error.OneginiRefreshMobileAuthPushTokenError;
 
 public class FCMInstanceIdService extends FirebaseInstanceIdService {
 
+  private static final String TAG = FCMInstanceIdService.class.getSimpleName();
+
   /**
-   * Called if InstanceID token is updated. This may occur if the security of the previous token had been compromised. This call is initiated by the InstanceID
-   * provider.
+   * Called if InstanceID token is created or updated. This may occur if the security of the previous token had been compromised.
+   * This call is initiated by the InstanceID provider.
    */
   @Override
   public void onTokenRefresh() {
-    final OneginiMobileAuthWithPushEnrollmentHandler mobileAuthWithPushEnrollmentHandler = new OneginiMobileAuthWithPushEnrollmentHandler() {
+    final String refreshedToken = FirebaseInstanceId.getInstance().getToken();
+    Log.d(TAG, "Refreshed token: " + refreshedToken);
+
+    final FCMRegistrationService fcmRegistrationService = new FCMRegistrationService(this);
+    fcmRegistrationService.updatePushToken(refreshedToken, new OneginiRefreshMobileAuthPushTokenHandler() {
       @Override
       public void onSuccess() {
-
+        Log.d(TAG, "The token has been updated: " + refreshedToken);
```

Comment:
The Example App is for developers, so showing them logs with additional information is ok. Especially in a service that has no UI to provide a proper feedback.